sources/packages/backend/libs/sims-db/src/entities/application-offering-change-request.model.ts

```diff
@@ -0,0 +1,136 @@
+import {
+  Column,
+  Entity,
+  JoinColumn,
+  ManyToOne,
+  OneToOne,
+  PrimaryColumn,
+} from "typeorm";
+import { ColumnNames, TableNames } from "../constant";
+import { RecordDataModel } from "./record.model";
+import { Application } from "./application.model";
+import { EducationProgramOffering } from "./education-program-offering.model";
+import { User } from "./user.model";
+import { Note } from "./note.model";
+import { ApplicationOfferingChangeRequestStatus } from "./application-offering-change-request-status.type";
+
+const REASON_MAX_LENGTH = 500;
+/**
+ * Represents the list of application specific offering change request,
+ * which is requested by institution.
+ */
+@Entity({ name: TableNames.ApplicationOfferingChangeRequests })
+export class ApplicationOfferingChangeRequest extends RecordDataModel {
+  /**
+   * Auto-generated sequential primary key column.
+   */
+  @PrimaryColumn({
```

Comment:
We can go with @PrimaryGeneratedColumn() which is more accurate.

Our application has a mix of both @PrimaryGeneratedColumn() and @PrimaryColumn().

I see that all the usages @PrimaryColumn() are either very old code or the primary key is not auto-generated.

Reply:
ok..I was not aware of that. updated